### DISCUSSION TITLE
How to get last 30 minutes active users?

### DISCUSSION THREAD

**Discussion by TharukaDananjaya** [freekmurze/laravel-analytics#515] [Q&A] [answered]
I want to get last 30 minutes or want to get minutes range data : 
`public function usersInLast30Minutes($propery_id)
     {
        try {
            Config::set('analytics.property_id', $propery_id);

            $startTime = Carbon::now()->subMinutes(30);
            $endTime = Carbon::now();
            $period = Period::create($startTime, $endTime);
            return $this->getAnalyticsData($period, [
                'activeUsers'
            ], ['country']);
        } catch (\Throwable $th) {
            throw $th;
        }
    }

    /**-----------------------------------------------Common _--------------------------------- */

    public function getAnalyticsData($period, $metrics = [], $dimensions = [], $limit = 10)
    {

        // Get analytics data for each time range
        $analyticsData = [];
        try {
            $response = Analytics::get(
                $period,
                $metrics,
                $dimensions,
                $limit
            );

            return $response;
        } catch (InvalidPeriod $e) {
            // Handle invalid period exception
            throw $e;
        } catch (\Exception $e) {
            // Handle other exceptions
            throw $e;
        }

        return $analyticsData;
    }`
Above method not work. i think its only work for date ranges not time ranges. Anyone know how to do that?

**Comment by jaroslavstefanec** [comment]
Hey,
The latest version, 5.5.0, introduces a new method, `getRealtime`, which allows you to retrieve real-time data from GA4.
```
use Spatie\Analytics\Facades\Analytics;
use Spatie\Analytics\Period;

// number of active users in the last 30 minutes which is visible in GA4 -> reports -> realtime overview
$period = Period::create(Carbon::today(), Carbon::today());
Analytics::getRealtime(period: $period, metrics: ['activeUsers']);
```

**Reply by Perseus24** [reply]
Hello, 
When I execute the above code, I am getting this error

`Spatie\Analytics\AnalyticsClient::runRealtimeReport(): Return value must be of type Google\Analytics\Data\V1beta\RunRealtimeReportResponse, Google\Analytics\Data\V1beta\RunReportResponse returned`

**Reply by jaroslavstefanec** [reply, marked as answer]
Hello, a pull request has been created: https://github.com/spatie/laravel-analytics/pull/536. If it is merged, you should adjust the period to, for example
```php
$period = Period::create(Carbon::now()->subMinutes(29), Carbon::now()); 
``` 

Please be aware that the maximum time allowed in the past is set to 29 minutes, otherwise, Google methods will throw an exception.

**Reply by tcgumus** [reply]
it works for the first 30 minutes but gives error when time passes 30 minute mark.

`{ "message": "endMinutesAgo must be less than or equal to startMinutesAgo. The API received startMinutesAgo = 29 and endMinutesAgo = 46", "code": 3, "status": "INVALID_ARGUMENT", "details": [] }
`
